### ISSUE TITLE
how to modify face detection and tracking algorithm for cartoon faces?

### ISSUE THREAD

**Issue by saumyaborwankar** [TaoRuijie/TalkNet_ASD#3]
I want to apply this repo to a cartoon video but it is not able to detect cartoon faces. What can be the solution? Should I retrain again with the data (in AVA format) or just modify the face detection algorithm?

**Comment by TaoRuijie** [comment]
1. The reason is that this face detection model can not detect the cartoon face, this face detection model can only detect human face I think.

2. I think the TalkNet model can not perform well in cartoon faces unless you have trained it in the cartoon dataset. Because the pretrain-model is trained only in the human face, it has no generation ability for the cartoon faces.

**Comment by saumyaborwankar** [comment]
I have a separate face detection model for my cartoon data. How can I integrate that with the rest of the code?

**Comment by TaoRuijie** [comment]
Do you have the active speaker detection label for your cartoon data? 

If so, assume your data is the face clips, saved in the image format (frame-by-frame). You can only modify the dataLoader.py file to load these images and the labels. The rest code is the same.

**Comment by saumyaborwankar** [comment]
I dont want to train the code again I just want my face detection model predictions (bounding boxes) to go forward with the audio.

**Comment by TaoRuijie** [comment]
Sorry I can not understand. Can you describe it more clearly? What do you want to achieve and which step is your problem?

**Comment by saumyaborwankar** [comment]
I want to achieve ASD without training any models. 
I have active speaker label for my data (audio timestamps) but I do not have bounding boxes for those videos
I already have a trained face detection model (for my cartoon data) which outputs bounding boxes present in a frame.
I want to use these bounding boxes to get the labels for active speakers in the cartoon video.

**Comment by TaoRuijie** [comment]
As for my understanding, you have the ASD label for your cartoon data already, and you can extract the bounding boxes by your face detection model.

For the face detection and forward audio and face images part, you can check the 'demoTalkNet.py' code.

`inference_video` function is for face detection, you can replace it with your face detection model
`track_shot` function is to do face tracking
`crop_video` function is to crop the detected face tracks
`evaluate_network` function is to forward audio and face images.

**Comment by saumyaborwankar** [comment]
There is another problem now with the scene detection. Its not able to detect scenes. Any suggestions?

**Comment by TaoRuijie** [comment]
Sorry, I am not mainly working in this scene detection. I only try it in some vlogs on youtube. So you can try another scene detection algorithm if this one can not meet your requirement.

**Comment by saumyaborwankar** [comment]
Thanks for all the information, could you tell me how you deal with frames that have no detected faces?

**Comment by TaoRuijie** [comment]
just skip them. If there is no face there is no need to do active speaker detection task.

**Comment by saumyaborwankar** [comment]
But do you still include them in the faces variable or you dont add them?
`# Face detection for the video frames`
`faces = inference_video(args)`

**Comment by TaoRuijie** [comment]
In `faces = inference_video(args)`. Outputs are the detected faces. 

The format for each data is :{'frame':xxx, 'bbox':xxx, 'conf':xxx}. You can try and print it, then you will know the format.

**Comment by saumyaborwankar** [comment]
No thats not what im asking, for example if there are no faces in frame 30, should the faces variable be 
`faces[0]={'frame':29, 'bbox':[1,2,3,4], 'conf':0.9}`
`faces[1]={'frame':30, 'bbox':[0,0,0,0] 'conf':0}`
`faces[2]={'frame':31, 'bbox':[3,4,5,6] 'conf':0.89}`
or should it be like this.
`faces[0]={'frame':29, 'bbox':[1,2,3,4], 'conf':0.9}`
`faces[1]={'frame':31, 'bbox':[3,4,5,6] 'conf':0.89}`

**Comment by TaoRuijie** [comment]
The second one

**Comment by saumyaborwankar** [comment]
Is there any way I can get a DER (Diarization error rate) for the output? Also what does the scores.pckl contain?

**Comment by TaoRuijie** [comment]
I am not familiar with DER, sorry can not help.

score.pckl contains ASD prediction result,
